internal/util/flowgraph/flow_graph.go

```diff
@@ -108,8 +118,9 @@ func (fg *TimeTickedFlowGraph) Close() {
 // NewTimeTickedFlowGraph create timetick flowgraph
 func NewTimeTickedFlowGraph(ctx context.Context) *TimeTickedFlowGraph {
 	flowGraph := TimeTickedFlowGraph{
-		nodeCtx: make(map[string]*nodeCtx),
-		closeWg: &sync.WaitGroup{},
+		nodeCtx:         make(map[string]*nodeCtx),
+		closeWg:         &sync.WaitGroup{},
+		closeGracefully: atomic.NewBool(CloseImmediately),
```

Comment:
default is CloseImmediately, why ?

Reply:
Only need to close gracefully when doing rolling upgrades, This is what the PR for. Gracefully will try to clean all buffered msgs in the flowgraph and update meta, we don't want to do it every time when we release a flowgraph, for example, when dropping collections, when balance channels.